### PR TITLE
✨ Prepare metal3-ipam as a CAPI IPAM provider

### PR DESCRIPTION
**Pull request by kashifest**

### CODE REVIEW

File: metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  contract: v1beta1
+- major: 1
+  minor: 3
+  contract: v1beta1
+- major: 1
+  minor: 2
+  contract: v1beta1
+- major: 1
+  minor: 1
+  contract: v1beta1
```
Comment (kashifest) [inline metadata.yaml:18]: @lentzi90 do you think we need to add the older contracts here? I already skipped the v1ax contracts, IMO adding older v1b1 contracts doesn't make much sense, wdyt?

Comment (lentzi90) [reply metadata.yaml:18]: Yes I think we can safely skip them. This looks good!

**Comment by Rozzii** [comment]
@kashifest  how time sensitive it is, would it be fine to rebae this after we have pushed DCO enforcement?

**Comment by kashifest** [comment]
> @kashifest how time sensitive it is, would it be fine to rebae this after we have pushed DCO enforcement?

Yes, it can wait
/hold

**Comment by kashifest** [comment]
This would require more changes, @peppi-lotta will take care of it in a seperate PR

**Pull request closed without merging**